biz.aQute.bndlib/src/aQute/bnd/osgi/Jar.java

```diff
@@ -432,6 +497,72 @@ public Map<String, Resource> getResources() {
 		return resources;
 	}
 
+	/**
+	 * returns an (unmodifiable) view of resources in this jar according to the
+	 * given release version.
+	 *
+	 * @return a map whose keys are resource names and the value the resource
+	 */
+	public Map<String, Resource> getVersionedResources(int release) {
+		if (isMultiRelease()) {
+			check();
+			Map<String, NavigableMap<Integer, Resource>> versionedResources = getAllVersionMap();
+			return versionedResources.entrySet()
+				.stream()
+				.map(versions -> {
+					Entry<Integer, Resource> releaseEntry = versions.getValue()
+						.headMap(release, true)
+						.lastEntry();
```

Comment:
I think that this method has the same issue as `getVersionedResource`, in that it fails to take account of all version folders lower than `release`, with higher versions taking precedence over lower versions.

Reply:
Do you "think" this or did you tried it out? Just curious, because `headMap` returns a map of all keys lower or equal to release and `lastEntry` returns the highest value from it, sounds exactly as what you explained here...

Reply:
I don't have a slot to check the code out and run it today but rather than letting the PR sit I did these checks by eyeballing the code. 
It's entirely possible that I've misread the data structure you're using and everything here is actually fine. If that is the case I would recommend some JavaDoc on `getAllVersionMap` so that others don't make the same mistake when maintaining the class in future.

Reply:
I'll enhance the javadoc and test-case so we can see at least that passing in the CI (it just requires an approval from a bnd committer as I'm a first-time contributor to this repository...).

Reply:
Done.